cli/tests/integration/npm_tests.rs

```diff
@@ -2219,3 +2219,240 @@ itest!(require_resolve_url_paths {
   cwd: Some("npm/require_resolve_url/"),
   copy_temp_dir: Some("npm/require_resolve_url/"),
 });
+
+#[test]
+pub fn byonm_cjs_esm_packages() {
+  let test_context = TestContextBuilder::for_npm()
+    .env("DENO_UNSTABLE_BYONM", "1")
+    .use_temp_cwd()
+    .build();
```

Comment:
The failing test on Windows is some symlink stuff. I can reproduce it locally by using `.use_symlinked_temp_dir()` in the `TestContextBuilder`. It's probably something small that shouldn't block reviewing this. Will fix it tomorrow.